sumologic/provider.go

```diff
@@ -134,22 +143,37 @@ func resolveRedirectURL(accessId string, accessKey string) (string, error) {
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	accessId := d.Get("access_id").(string)
 	accessKey := d.Get("access_key").(string)
+	authJwt := d.Get("auth_jwt").(string)
 	environment := d.Get("environment").(string)
 	baseUrl := d.Get("base_url").(string)
 	isInAdminMode := d.Get("admin_mode").(bool)
 
 	msg := ""
-	if accessId == "" {
-		msg = "sumologic provider: access_id should be set;"
-	}
-
-	if accessKey == "" {
-		msg = fmt.Sprintf("%s access_key should be set; ", msg)
+	if authJwt == "" {
+		if accessId == "" || accessKey == "" {
+			msg = "sumologic provider: auth_jwt is not set;"
+		}
+		if accessId == "" {
+			msg = fmt.Sprintf("%s access_id should be set;", msg)
+		}
+		if accessKey == "" {
+			msg = fmt.Sprintf("%s access_key should be set; ", msg)
+		}
+	} else {
+		if accessId != "" || accessKey != "" {
+			msg = "sumologic provider: auth_jwt is set;"
+		}
+		if accessId != "" {
+			msg = fmt.Sprintf("%s access_id should not be set;", msg)
+		}
+		if accessKey != "" {
+			msg = fmt.Sprintf("%s access_key should not be set; ", msg)
+		}
```

Comment:
I don't think that having access id / key + auth_jwt in the env variables should necessarily warrant an error response. For example, we might have both of them set in the environment and want to alternate between using JWT auth or the access id/key auth depending on the operation intended.

Reply:
Right -- removed this whole section, in accordance with changing access id/key back to `Required` (with dummy values to be passed in the JWT case). The logic is now: if JWT is not provided, have the old checks/messages for access id/key; and if it is, ignore them.